admiral/pkg/clusters/serviceentry_test.go

```diff
@@ -268,17 +278,78 @@ import (
 //	}
 //}
 //
-func buildFakeConfigMapFromAddressStore(addressStore *ServiceEntryAddressStore, resourceVersion string) *v1.ConfigMap{
-	bytes,_ := yaml.Marshal(addressStore)
+func buildFakeConfigMapFromAddressStore(addressStore *ServiceEntryAddressStore, resourceVersion string) *v1.ConfigMap {
+	bytes, _ := yaml.Marshal(addressStore)
 
 	cm := v1.ConfigMap{
 		Data: map[string]string{"serviceEntryAddressStore": string(bytes)},
 	}
-	cm.Name="se-address-configmap"
-	cm.Namespace="admiral-remote-ctx"
-	cm.ResourceVersion=resourceVersion
+	cm.Name = "se-address-configmap"
+	cm.Namespace = "admiral-remote-ctx"
+	cm.ResourceVersion = resourceVersion
 	return &cm
 }
+
+func TestModifyNonExistingSidecarForLocalClusterCommunication(t *testing.T) {
+	config, error := clientcmd.BuildConfigFromFlags("", "../test/resources/admins@fake-cluster.k8s.local")
+
+	if error != nil {
+		t.Errorf("%v", error)
+	}
+	stop := make(chan struct{})
+	handler := test.MockSidecarHandler{}
+
+	sidecarController, _ := istio.NewSidecarController(stop, &handler, config, time.Duration(1000))
+
+	remoteController := &RemoteController{SidecarController: sidecarController}
+
+	modifySidecarForLocalClusterCommunication("test-sidecar-namespace", "test-dependency-namespace", "test-local-fqdn", remoteController)
+
+	sidecarObj, _ := sidecarController.IstioClient.NetworkingV1alpha3().Sidecars("test-sidecar-namespace").Get(common.GetWorkloadSidecarName(), v12.GetOptions{})
+
+	if sidecarObj == nil || sidecarObj.Spec.Egress != nil {
```

Comment:
Can you be a little more specific with the assertion here? What should `sidecarObj.Spec.Egress` be?

Reply:
This will change based on what is configured in the workload namespace. For this test, assumption is that no sidecar resource is configured. So test is that we are not creating this resource.